### ISSUE TITLE
unsupported mode '2' for pin '15' at ../lib/Device/Firmata/Platform.pm line 823.

### ISSUE THREAD

**Issue by bcarroll** [ntruchsess/perl-firmata#15]
examples/test.pl is producing an error "unsupported mode '2' for pin '15' at ../lib/Device/Firmata/Platform.pm line 823." 

This test was done using an Arduino Mega 2560 R3 and the StandardFirmata sketch included with Arduino 1.0.5.

I did not get this error with an Arduino Nano v3.

The firmata.org WIKI mentions changing the samplingInterval to 99 to support all pins on the Arduino Mega, but I thought that had been included in the newer versions of the StandardFirmata sketch.

**Comment by jnsbyr** [comment]
Looks to me like the error message is correct for an Arduino Mega 2560 R3.

From your logging one can see that examples/test.pl:24 is executed successfully ($device->pin_mode(12,PIN_INPUT)) -> f4,0c,00 immediately followed by the error message that results from examples/test.pl:25 ($device->pin_mode(15,PIN_ANALOG)).

The Mega 2560 has a lot of IOs and pin 15 = D15 is a digital only pin. Check the Firmata board definition, the Perl Firmata pin number for A15 is probably 30. 

In other words: The test program does not check the hardware capabilities and must be adapted to the hardware in some cases.
